tests/phpunit/tests/post/query.php

```diff
@@ -777,4 +777,525 @@ public function test_found_posts_should_be_integer_even_if_found_posts_filter_re
+
+		// Add a mixture of meta values so all 5 posts match the meta query.
+		add_post_meta( $ids[0], 'field_1', 'foo' );
+		add_post_meta( $ids[0], 'field_2', 'bar' );
+		add_post_meta( $ids[1], 'field_1', 'foo' );
+		add_post_meta( $ids[2], 'field_1', 'foo' );
+		add_post_meta( $ids[3], 'field_2', 'bar' );
+		add_post_meta( $ids[4], 'field_2', 'bar' );
+
+		// This query results in a `GROUP BY wp_posts.ID` clause, which means the
+		// count query must count `DISTINCT wp_posts.ID` to eliminate duplicates.
+		$q = new WP_Query(
+			array(
+				'fields'         => $fields,
+				'posts_per_page' => 2,
+				'meta_query'     => array(
+					'relation' => 'OR',
+					array(
+						'key' => 'field_1',
+					),
+					array(
+						'key' => 'field_2',
+					),
+				),
+			)
+		);
+
+		$this->assertSame( 2, $q->post_count, self::get_count_message( $q ) );
+		$this->assertSame( 5, $q->found_posts, self::get_count_message( $q ) );
+		$this->assertEquals( 3, $q->max_num_pages, self::get_count_message( $q ) );
+	}
+
+	/**
+	 * @ticket 47280
+	 * @dataProvider data_fields
+	 *
+	 * @param string $fields Value of the `fields` argument for `WP_Query`.
+	 */
+	public function test_found_posts_are_correct_for_group_by_queries( $fields ) {
+		$user1 = self::factory()->user->create();
+		$user2 = self::factory()->user->create();
+
+		self::factory()->post->create_many(
+			5,
+			array(
+				'post_author' => $user1,
+			)
+		);
+		self::factory()->post->create_many(
+			5,
+			array(
+				'post_author' => $user2,
+			)
+		);
+
+		/**
+		 * Adds a GROUP BY clause to the query.
+		 */
+		add_filter(
+			'posts_groupby_request',
+			function() {
+				return "{$GLOBALS['wpdb']->posts}.post_author";
+			}
+		);
+
+		// This query results in a `GROUP BY wp_posts.post_author` clause, which means the
+		// count query must count `DISTINCT wp_posts.post_author` to eliminate duplicates.
+		$q = new WP_Query(
+			array(
+				'fields'         => $fields,
+				'posts_per_page' => 2,
+			)
+		);
+
+		$this->assertSame( 2, $q->post_count, self::get_count_message( $q ) );
+		// There is a total of two distinct authors in the results.
+		$this->assertSame( 2, $q->found_posts, self::get_count_message( $q ) );
+		$this->assertEquals( 1, $q->max_num_pages, self::get_count_message( $q ) );
+	}
+
+	/**
+	 * @ticket 47280
+	 * @group ms-required
+	 * @dataProvider data_fields
+	 *
+	 * @param string $fields Value of the `fields` argument for `WP_Query`.
+	 */
+	public function test_found_posts_are_correct_when_switching_between_sites( $fields ) {
+		$blog = self::factory()->blog->create();
+		self::factory()->post->create_many( 5 );
+
+		$q = new WP_Query(
+			array(
+				'fields'         => $fields,
+				'posts_per_page' => 2,
+			)
+		);
+
+		// Switch to another site.
+		switch_to_blog( $blog );
+
+		// Count the posts from the original site. This works because the SQL query
+		// and its table names has already been formed during the original query.
```

Comment:
```
/*
 * Multi-line comments are formatted like
 * this according to the coding standards
 */
```

:1234: Note: This comment applies in multiple locations but I've only noted it once.